pkg/serving/config_changes.go

```diff
@@ -331,14 +333,37 @@ func UpdateContainerArg(template *servingv1.RevisionTemplateSpec, arg []string)
 	return nil
 }
 
-// UpdateContainerPort updates container with a give port
-func UpdateContainerPort(template *servingv1.RevisionTemplateSpec, port int32) error {
+// UpdateContainerPort updates container with a given name:port
+func UpdateContainerPort(template *servingv1.RevisionTemplateSpec, port string) error {
 	container, err := ContainerOfRevisionTemplate(template)
 	if err != nil {
 		return err
 	}
+	var containerPort int64
+	var name string
+
+	if strings.Contains(port, ":") {
+		portDetails := strings.Split(port, ":")
```

Comment:
```suggestion
		portDetails := strings.SplitN(port, ":", 2)
```

This will split in at most 2 elements. If there are more `:` then this will lead to a parse error later as the second element is not an int. So you have automatically a validation here.